### PR TITLE
Genrebranch

### PR DESCRIPTION
**Pull request by SpencerMilner**

### CODE REVIEW

**Comment by lava10** [comment]
💃 💃 💃 💃 💃

File: TeenTixMobileNEW/EventsByGenreTableViewController.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			this.Title = catName;
+
+			this.TabBarController.TabBar.TintColor = UIColor.White;
+
+			ParentViewController.NavigationController.SetNavigationBarHidden (true, true);
+			NavigationItem.SetRightBarButtonItem(
+				new UIBarButtonItem("Sign Out", UIBarButtonItemStyle.Plain, (object sender, EventArgs e) => {
+					ParentViewController.NavigationController.DismissViewController(true, null);
+					RestAPI.Session = null;
+				}), true);
```
Comment (thomasvandoren) [inline TeenTixMobileNEW/EventsByGenreTableViewController.cs]: I think we can remove lines 31-38 here. We only need those set on the home view controller.

File: TeenTixMobileNEW/EventsByGenreTableViewController.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+			var events = Database.DB.GetEventsByCategoryStartingAtDate (DateTime.Today, categoryID);
+			tableSource = new EventTableSource<EventsByGenreTableViewController> (events, this, "EventsByCatTableCellId");
+			TableView.Source = tableSource;
+
+			await UpdateEvents ();
```
Comment (thomasvandoren) [inline TeenTixMobileNEW/EventsByGenreTableViewController.cs]: Let's skip updating events on this controller. We do it in several other places.

File: TeenTixMobileNEW/EventsByGenreTableViewController.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		private async Task<bool> UpdateEvents() {
+			Console.WriteLine ("Updating events...");
+			await RestAPI.UpdateCalendar (DateTime.Today);
+			var events = Database.DB.GetEventsStartingAtDate (DateTime.Today);
+			Console.WriteLine ("Updating events on HomeTableSource");
+			tableSource.UpdateEvents (events);
+			return true;
+		}
+
+		private async Task<bool> GetFavorites() {
```
Comment (thomasvandoren) [inline TeenTixMobileNEW/EventsByGenreTableViewController.cs]: Is this method used? If not, let's remove it.

File: TeenTixMobileNEW/GenreTableViewController.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			this.Title = "Explore";
+
+			this.TabBarController.TabBar.TintColor = UIColor.White;
+
+			ParentViewController.NavigationController.SetNavigationBarHidden (true, true);
+			NavigationItem.SetRightBarButtonItem(
+				new UIBarButtonItem("Sign Out", UIBarButtonItemStyle.Plain, (object sender, EventArgs e) => {
+					ParentViewController.NavigationController.DismissViewController(true, null);
+					RestAPI.Session = null;
+				}), true);
```
Comment (thomasvandoren) [inline TeenTixMobileNEW/GenreTableViewController.cs]: Same comment here.

File: TeenTixMobileNEW/GenreTableViewController.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+			var events = Database.DB.getCategories (DateTime.Today);
+			tableSource = new CategoryTableSource<GenreTableViewController> (events, this, "GenreTableCellId");
+			TableView.Source = tableSource;
+
+			await UpdateEvents ();
```
Comment (thomasvandoren) [inline TeenTixMobileNEW/GenreTableViewController.cs]: Same here.

**Comment by thomasvandoren** [comment]
LGTM ✨

**Comment by thomasvandoren** [comment]
👍

**Pull request merged**